index out of range when syncing Görli

`build/bin/tg --goerli --private.api.addr=localhost:9090`

commit 904a992d3cfedfb883a8aafcd86a922b123af466 (HEAD -> master, origin/master, origin/HEAD)

```
INFO [03-01|18:10:58.687] Timings                                  Headers=50.450861ms    BlockHashes=1.836088ms    Bodies=3.738011ms   Bodies="511.364µs"  Senders=5.912103ms   Execution=10.336065ms  HashState=4.69714ms    IntermediateHashes=7.438001ms   AccountHistoryIndex=1.467774ms   StorageHistoryIndex=1.18364ms    LogIndex="747.42µs"  TxLookup="288.831µs" TxPool="499.237µs" Finish="20.965µs"
INFO [03-01|18:10:58.698] Commit cycle                             in=10.253951ms
panic: runtime error: index out of range [-1]

goroutine 67 [running]:
github.com/ledgerwatch/turbo-geth/consensus/clique.(*Verifier).snapshot(0xc000122028, 0xc010c8b108, 0x1, 0x2, 0x8, 0xb4aa4c25d31ec401, 0xb4007fbb80731200)
	github.com/ledgerwatch/turbo-geth/consensus/clique/verifier.go:227 +0xcaa
github.com/ledgerwatch/turbo-geth/consensus/clique.(*Verifier).verifyCascadingFields(0xc000122028, 0xc0030af680, 0xc010c8b108, 0x1, 0x2, 0x43bc29, 0xc0003306c0)
	github.com/ledgerwatch/turbo-geth/consensus/clique/verifier.go:126 +0x8f
github.com/ledgerwatch/turbo-geth/consensus/clique.(*Verifier).verifyHeader(0xc000122028, 0xc0030af680, 0xc010c8b108, 0x1, 0x2, 0x1fd22a0, 0x4, 0x100000044b01b)
	github.com/ledgerwatch/turbo-geth/consensus/clique/verifier.go:114 +0x311
github.com/ledgerwatch/turbo-geth/consensus/clique.(*Verifier).Verify(0xc000122028, 0x17e3260, 0x1fd22a0, 0xc0030af680, 0xc010c8b108, 0x1, 0x2, 0xc019fa0100, 0xc000469c28, 0xc000582d80)
	github.com/ledgerwatch/turbo-geth/consensus/clique/verifier.go:27 +0x78
github.com/ledgerwatch/turbo-geth/consensus/process.(*Consensus).verifyByRequest(0xc0003c6420, 0x49af3d0dd508f48a, 0xc0030af680, 0x1, 0x1, 0xc010c8b108, 0x1, 0x2, 0x2fff90d26ab808c, 0xab7233ac91ad1526)
	github.com/ledgerwatch/turbo-geth/consensus/process/process.go:278 +0xb4
github.com/ledgerwatch/turbo-geth/consensus/process.NewConsensusProcess.func1(0xc0003c6420, 0xc000386300)
	github.com/ledgerwatch/turbo-geth/consensus/process/process.go:118 +0x104d
created by github.com/ledgerwatch/turbo-geth/consensus/process.NewConsensusProcess
	github.com/ledgerwatch/turbo-geth/consensus/process/process.go:50 +0xe5
```

--- a/consensus/clique/verifier.go
+++ b/consensus/clique/verifier.go
@@ -224,7 +224,11 @@ func (c *Verifier) snapshot(parents []*types.Header) (*Snapshot, error) {
 	}
 
 	if snap == nil {
-		return nil, fmt.Errorf("a nil snap for %d block: %w", parents[i].Number.Uint64(), consensus.ErrUnknownAncestor)
+		if len(parents) > 0 {
+			return nil, fmt.Errorf("a nil snap for %d block: %w", parents[i].Number.Uint64(), consensus.ErrUnknownAncestor)
+		}
+
+		return nil, consensus.ErrUnknownAncestor
 	}
 
 	if len(parents) > 0 {

